File: 26-03-18/java/02-The-Football-Fest.java

import java.io.*;
import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(input.readLine());
        while(t-- > 0)
        {
            int n, id;
            String str[] = input.readLine().split(" ");
            n = Integer.parseInt(str[0]);
            id = Integer.parseInt(str[1]);
            Stack<Integer> st = new Stack<Integer>();
            st.push(id);
            int last = 0;
            while(n-- > 0)
            {
                String str1[] = input.readLine().split(" ");
                if(str1[0].equals("P") )
                {
                    
                    id = Integer.parseInt(str1[1]);
                    st.push(id);
                    last = 0;
                }
                else
                {
                    if(!st.empty())
                    {
                        if(last == 0)
                        {
                            last = st.peek();
                            st.pop();
                        }
                        else
                        {
                            int tmp = last;
                            last = st.peek();
                            st.pop();
                            st.push(tmp);
                        }
                    }
                }
            }
            System.out.println("Player " + st.peek() );
        }
        
    }
}

File: 26-03-18/python/02-The-Football-Fest.py
for _ in range( int(input()) ):
    n, idd = map(int, input().split())
    st = []
    st.append(idd)
    last = 0
    for __ in range(0, n):
        ch = input()
        if(ch.split()[0] == 'P'):
            idd = int(ch.split()[1])
            st.append(idd)
            last = 0
        else:
            i = len(st) - 1
            if(i > 0):
                if(last == 0):
                    last = st[i];
                    del st[i]
                else:
                    tmp = last
                    last = st[i]
                    del st[i]
                    st.append(tmp)
    print("Player",st[len(st) - 1])
